packages/client-proxy/main.go

```diff
@@ -138,9 +140,28 @@ func main() {
 	}()
 
 	// Proxy request to the correct node
-	server := http.Server{Addr: fmt.Sprintf(":%d", port)}
+	server := &http.Server{Addr: fmt.Sprintf(":%d", port)}
 	server.Handler = http.HandlerFunc(proxy(logger))
 
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
```

Comment:
also cancel the root context in a defer here so that we don't block waiting for a sigterm waiting for the health endpoint if there's a crash or problem with this service.